advanced/Exercise_7.md

```diff
@@ -1,154 +1,147 @@
+> **Hint:** Don't forget that all scripts need to be executable! 
 
-```plaintext
-cd ..
-git status
-```
+Finally, verify that both *pre-commit* hooks are working correctly.
 
-Git is now informing us that we now have modified content in the posters submodule.  
+## Local Testing with the `pre-commit` Tool <a name="pre-commit"></a>
 
-## Create new content locally and update the sub-repository <a name="push"></a>
-Now, let's commit the change to the posters submodule, and push that change to our posters repository on Github.
+In order to manually run the pre-commit hooks to preview your actions without committing, you can utilize the [pre-commit](https://pre-commit.com/) tool.
 
-Go back into the posters submodule and add and commit the change we just made.
+If you haven't already, install the `pre-commit` tool globally on your system:
 
-```plaintext
-cd posters
-git commit -am "Add poster 1 to the schedule"
+```sh
+pip install pre-commit
 ```
 
-You can treat the submodule like any git repository and simply push the new commit to your fork on Github.  
+or
 
-```plaintext
-git push origin
+```sh
+conda install -c conda-forge pre-commit
 ```
 
-If you navigate back to your Github fork (and refresh the page if necessary), you should see the commit that you just made.  
+After installation, you can verify that pre-commit is available by running:
 
-The conference planning repository now needs to be updated, to point to the new commit in the posters submodule.  Let's do this now.
-
-```plaintext
-cd ..
-git add posters
-git commit -m "Update posters submodule"
+```sh
+pre-commit --version
 ```
 
-Now, if you check the submodule status, you will see that it is pointing to the latest commit you made.
+This should display the version of `pre-commit`, confirming a successful installation.
 
-## Create new content in the sub-repository and update the local repository <a name="pull"></a>
+In your *conference_planning* repository, create a *.pre-commit-config.yaml* file to configure which hooks to run. Here is an example configuration:
 
-Finally, let's learn how to pull changes from the submodule into our host repository.  
+```yaml
+repos:
+- repo: local
+  hooks:
+  - id: pre-commit-whitespace
+    name: check for trailing whitespaces
+    entry: .git/hooks/pre-commit-whitespace
+    language: system
+  # Add other hooks here if needed
+```
 
-Navigate back to your fork of the poster repository on Github, and make a change to the schedule file and commit it. You can do this directly on the web interface by simply selecting the file and using the edit button.
+This configuration specifies that the `pre-commit-whitespace` hook (the one we created earlier) should run.
 
-Once you have done this, go to back to the terminal and use `git submodule update` to get the main repository to fetch the new commit of the poster repository.
+To manually run the `pre-commit` checks without making a commit, use the following command:
```

Comment:
It should be mentioned that this command only affects staged files.

Reply:
Done 👍🏻